### ISSUE TITLE
BlenderBIM - Bug in migrator utility (IFC4x3)

### ISSUE THREAD

**Issue by mdjska** [IfcOpenShell/IfcOpenShell#2857]
When trying to load a classification in a IFC4x3 model, I get an error that ends with:

```
File ".../Application Support/Blender/3.4/scripts/addons/blenderbim/libs/site/packages/ifcopenshell/util/schema.py", line 286, in migrate_attribute
    if value is None and not attribute.optional():
UnboundLocalError: local variable 'value' referenced before assignment
```

**Comment by Moult** [comment]
@mdjska would you like to try and help debug and fix this one?

**Comment by mdjska** [comment]
I could try, but my timeline on this might be a bit long.. But I took a look at the schema.py file yesterday though, and is it right that these mapping files (like the "class_4_to_2x3.json" and "class_2x3_to_4.json") would also have to be made for 4X3?

**Comment by Moult** [comment]
@mdjska indeed, would you like to try? In general it can start as a blank file.

**Comment by mdjska** [comment]
Sorry, I'm really swamped this and next week, but if nobody else has had time to look at it until then, I can try in my easter break (in two weeks)

**Comment by Moult** [comment]
@mdjska how'd your easter break go? :)

**Comment by Andrej730** [comment]
It shouldn't fail completely after 39ee9cd. And if there is .ifc example we can add an attribute to the ifc4x3<->ifc4 attribute mapping as in 8728fa3

**Comment by sboddy** [comment]
@mdjska Can this be closed since Andrej's commits?

**Comment by sboddy** [comment]
Closing due to lack of response for more than 2 months. Please reopen if still an issue.
